alerts/cluster-monitoring-operator/PrometheusScrapeBodySizeLimitHit.md

```diff
@@ -0,0 +1,111 @@
+namespace `openshift-monitoring`.
+
+To check the value in configmap, use this command:
+```bash
+oc get cm -n openshift-monitoring cluster-monitoring-config -o yaml | grep enforcedBodySizeLimit
+```
+
+If the value is too small, we need to set it to a larger number. Please refer to
+[Mitigation](#mitigation) on how to change the value.
+
+If the value is correct, we need to find out which target is responding with a body
+too large for scraping.
+
+In the Openshift Web UI, go to the Observe > Targets page and check which
+targets are down.
+
+If we need more information than the UI provides, we can query the Prometheus API
+endpoint `/api/v1/targets`.
+In the field `data.activeTargets` search for those that have `health` field
+different than `up` and check for the `lastError` field for confirmation:
+
+1. Get the name of the secret containing the token of service account `prometheus-k8s`.
+We use the following command and check for the name  `prometheus-k8s-token-[a-z]+`
+in the line `Tokens`. In the example below, the secret name is `prometheus-k8s-token-nwtrf`.
+
+```bash
+$ oc describe sa prometheus-k8s -n openshift-monitoring
+Name:                prometheus-k8s
+Namespace:           openshift-monitoring
+Labels:              app.kubernetes.io/component=prometheus
+                     app.kubernetes.io/instance=k8s
+                     app.kubernetes.io/name=prometheus
+                     app.kubernetes.io/part-of=openshift-monitoring
+                     app.kubernetes.io/version=2.35.0
+Annotations:         serviceaccounts.openshift.io/oauth-redirectreference.prometheus-k8s:
+                       {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"prometheus-k8s"}}
+Image pull secrets:  prometheus-k8s-dockercfg-6wl6b
+Mountable secrets:   prometheus-k8s-dockercfg-6wl6b
+Tokens:              prometheus-k8s-token-nwtrf
+Events:              <none>
+```
+
+2. Decode the token from the secret.
+
+```bash
+token=$(oc get secret $secret_name_here -n openshift-monitoring -o jsonpath={.data.token} | base64 -d)
+```
+
+3. Get the route URL of the Prometheus API endpoint.
+
+```bash
+host=$(oc get route -n openshift-monitoring prometheus-k8s -o jsonpath={.spec.host})
+```
+
+4. List all the scraped targets with health status different than "up".
+
+```bash
+curl -H "Authorization: Bearer $token" -k https://${host}/api/v1/targets | jq '.data.activeTargets[]|select(.health!="up")'
+```
+
+5. Check the failling target's response body size. We simulate a scrape of its
+`scrapeUrl` using the command below and check for the body size.
+
+```bash
+oc exec -it prometheus-k8s-0 -n openshift-monitoring  -- curl -k --key /etc/prometheus/secrets/metrics-client-certs/tls.key --cert /etc/prometheus/secrets/metrics-client-certs/tls.crt --cacert /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt $scrape_url | wc --bytes
+```
+
+## Mitigation
```

Comment:
I see 2 options:
* Increase the limit
* Report a bug if you think that the target returns too much data.